File: src/Budgetify.Application/Accounts/AddAccount/NewAccountData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Budgetify.Domain.Accounts;
using Money.DB;

namespace Budgetify.Application.Accounts.AddAccount
{
    public sealed record NewAccountData(AccountType Type, AccountName Name, Currency Currency, decimal InitialBalance)
    {
    }
}

File: src/client-app/src/models/requests/newAccountData.ts
import {Money} from "../shared/money";

export interface NewAccountData {
    type: string
    name: string
    initialBalance: number
}

export class NewAccountData implements NewAccountData{
    constructor() {
        this.type = ''
        this.name = ''
        this.initialBalance = 0
    }
}